include/deal.II/differentiation/sd/symengine_scalar_operations.h

```diff
@@ -605,7 +914,176 @@ namespace Differentiation
+          add_to_symbol_map<ignore_invalid_symbols>(symbol_map,
+                                                    Expression(it->first));
+        }
+    }
+
+
+    template <bool ignore_invalid_symbols,
+              typename SymbolicType,
+              typename... Args>
+    void
+    add_to_symbol_map(types::substitution_map &symbol_map,
+                      const SymbolicType &     symbol,
+                      const Args &... other_symbols)
+    {
+      add_to_symbol_map<ignore_invalid_symbols>(symbol_map, symbol);
+      add_to_symbol_map<ignore_invalid_symbols>(symbol_map, other_symbols...);
+    }
+
+
+    template <typename ExpressionType, typename ValueType, typename>
+    void
+    set_value_in_symbol_map(types::substitution_map &substitution_map,
+                            const ExpressionType &   symbol,
+                            const ValueType &        value)
+    {
+      // Call the above function
+      using SE_RCP_Basic = const SymEngine::RCP<const SymEngine::Basic> &;
+      internal::set_value_in_symbol_map(substitution_map,
+                                        static_cast<SE_RCP_Basic>(symbol),
+                                        static_cast<SE_RCP_Basic>(
+                                          ExpressionType(value)));
+    }
+
+
+    template <typename SymbolicType, typename ValueType>
+    void
+    set_value_in_symbol_map(types::substitution_map &        substitution_map,
+                            const std::vector<SymbolicType> &symbols,
+                            const std::vector<ValueType> &   values)
+    {
+      Assert(symbols.size() == values.size(),
+             ExcDimensionMismatch(symbols.size(), values.size()));
+
+      typename std::vector<SymbolicType>::const_iterator it_symb =
+        symbols.begin();
+      typename std::vector<ValueType>::const_iterator it_val = values.begin();
+      for (; it_symb != symbols.end(); ++it_symb, ++it_val)
+        {
+          Assert(it_val != values.end(), ExcInternalError());
+          set_value_in_symbol_map(substitution_map, *it_symb, *it_val);
+        }
+    }
+
+
+    template <typename SymbolicType, typename ValueType>
+    void
+    set_value_in_symbol_map(
+      types::substitution_map &                 substitution_map,
+      const std::pair<SymbolicType, ValueType> &symbol_value)
+    {
+      set_value_in_symbol_map(substitution_map,
+                              symbol_value.first,
+                              symbol_value.second);
+    }
+
+
+    template <typename SymbolicType, typename ValueType, typename... Args>
+    void
+    set_value_in_symbol_map(
+      types::substitution_map &                 substitution_map,
+      const std::pair<SymbolicType, ValueType> &symbol_value,
+      const Args &... other_symbol_values)
+    {
+      set_value_in_symbol_map(substitution_map, symbol_value);
+      set_value_in_symbol_map(substitution_map, other_symbol_values...);
```

Comment:
what happens if `Args...` is empty? Or can this not happen?

Reply:
I believe that this should not happen (when the parameter pack is reduced to one argument, then one of the other functions should be called and the recursion is terminated).